pandas/src/period.pyx

```diff
@@ -458,13 +458,39 @@ def extract_ordinals(ndarray[object] values, freq):
 
     for i in range(n):
         p = values[i]
-        ordinals[i] = p.ordinal
-        if p.freqstr != freqstr:
-            msg = _DIFFERENT_FREQ_INDEX.format(freqstr, p.freqstr)
-            raise IncompatibleFrequency(msg)
+
+        if p is tslib.NaT:
+            ordinals[i] = tslib.iNaT
```

Comment:
I see that this is possible to be called with both int64 and object types so for sure need to check all cases here